src/group_theory/free_group.lean

```diff
@@ -863,4 +884,64 @@ fintype.subtype (list.to_finset $ red.enum L₁) $
 
 end reduce
 
+section metric
+
+variable [decidable_eq α]
+
+/-- The length of reduced words provides a norm on a free group. --/
+def norm (x : free_group α) : nat := x.to_word.length
+
+lemma norm_inv_le (x : free_group α) : norm x⁻¹ ≤ norm x :=
+begin
+  let w := x.to_word,
+ 
+  calc norm x⁻¹ = norm ((mk (x.to_word))⁻¹) : by simp only [to_word.mk]
+       ... = norm (mk (inv_rev w)) : by simp only [←inv_mk]
+       ... ≤ (inv_rev w).length : by exact to_word.le_length
+       ... = w.length : inv_rev_length
+       ... = norm x : rfl
+end
+
+lemma norm_inv_eq (x : free_group α) : norm x = norm x⁻¹ :=
+begin
+  refine le_antisymm _ (norm_inv_le x),
+  simpa using norm_inv_le x⁻¹
+end
+
+lemma norm_zero_eq_one (x : free_group α ) : norm x = 0 → x = 1 := 
```

Comment:
Would it make sense to actually make this ↔? Or to have the reverse lemma too?

Reply:
Yes, this should be an `iff`. Also, there should be `to_word_eq_nil_iff`.

Reply:
Thank you @laurentbartholdi and @urkud !  In 95020706117e13a733f24388199b05a70c71eb77, I've replaced `norm_zero_eq_one` with `norm_zero_iff_one` whose proof is now in terms of a new `to_word_eq_nil_iff`.